### PR TITLE
feat: add resolver config

### PR DESCRIPTION
**Pull request by Septimus4**
implement ENG-4408

# Checklist

- [x] `make test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/indykite/.github/blob/master/commit-message.md)
- [ ] is breaking change

## Description of change
Adds support for ExternalDataResolver

### CODE REVIEW

**Comment by Septimus4** [comment]
@arxeiss interesting our header obfuscation brakes Terraform 
```
 # indykite_external_data_resolver.development will be updated in-place
    ~ resource "indykite_external_data_resolver" "development" {
          id                = "gid:AAAAAAAAAAAAAAAAAAAAAAAAAAA"
          name              = "my-first-external-data-resolver1"
          # (13 unchanged attributes hidden)

        ~ headers {
            ~ name   = "Content-Type" -> "Authorization"
            ~ values = [
                ~ "application/json" -> "***",
              ]
          }
        ~ headers {
            ~ name   = "Authorization" -> "Content-Type"
            ~ values = [
                ~ "***" -> "application/json",
              ]
          }
      }
```
I have to found a way of handling sync with 
```
const (
	// SecretPlaceholder specify placeholder for secret non-empty value.
	SecretPlaceholder = "********"
	// SecretEmptyPlaceholder specify placeholder for secret empty value.
	SecretEmptyPlaceholder = "(empty)"
	// SecretUnchangedPlaceholder specify placeholder when old and new values are the same.
	SecretUnchangedPlaceholder = "(unchanged)"
	// SecretChangedPlaceholder specify placeholder when old and new values are different.
	SecretChangedPlaceholder = "(changed value)"
)
```

File: indykite/resource_consent.go [hunk 1]
```diff
@@ -104,17 +106,52 @@ func resourceConsentFlatten(
...
+			return 0
+		}
+		return uint64(v)
+	case uint64:
+		return v
+	default:
+		addError(diagnostics, "Invalid Type", "unsupported type for "+key+": "+reflect.TypeOf(v).String())
+		return 0
+	}
+}
```
Comment (maaland) [inline indykite/resource_consent.go:157]: nit: would have preferred for this to be in a separate commit, since it's not related to external data resolvers

File: indykite/resource_external_data_resolver.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+var (
+	externalDataResolverHeaderRegex = regexp.MustCompile(`^[a-zA-Z0-9-]+$`)
+)
+
+func resourceExternalDataResolver() *schema.Resource {
+	readContext := configReadContextFunc(resourceExternalDataResolverFlatten)
+
+	return &schema.Resource{
+		Description: "ExternalDataResolver is a configuration node that allows to fetch data from external sources",
```
Comment (maaland) [inline indykite/resource_external_data_resolver.go:46]: is it a `node`?

File: indykite/resource_external_data_resolver.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+					validation.IsURLWithHTTPorHTTPS,
+				),
+				Description: "Full URL to endpoint that will be called",
+			},
+			externalDataResolverMethodKey: {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringIsNotEmpty,
+					validation.StringInSlice([]string{"GET", "POST", "PUT", "PATCH"}, false),
```
Comment (maaland) [inline indykite/resource_external_data_resolver.go:82]: question: we allow `POST`, `PUT` and `PATCH`?

File: indykite/resource_external_data_resolver.go (hunk 2 above)
Comment (Septimus4) [reply indykite/resource_external_data_resolver.go:46]: No thank you, I am so old I still think config nodes are nodes

File: indykite/resource_consent.go (hunk 1 above)
Comment (Septimus4) [reply indykite/resource_consent.go:157]: true but the linter woke up and started to complain here, so the linter failed on the PR. I would guess we updated the linter or its config recently

File: indykite/resource_external_data_resolver.go (hunk 3 above)
Comment (Septimus4) [reply indykite/resource_external_data_resolver.go:82]: currently we don't really have the fetcher implemented so not sure what we will support. I will leave it like that and maybe come back to update it when backend is implemented

File: indykite/resource_external_data_resolver.go (hunk 2 above)
Comment (maaland) [reply indykite/resource_external_data_resolver.go:46]: unfortunately a lot of the public code still says "node" and we can't change that without making breaking changes..

File: indykite/resource_consent.go (hunk 1 above)
Comment (maaland) [reply indykite/resource_consent.go:157]: it's fine

File: indykite/resource_external_data_resolver.go (hunk 3 above)
Comment (maaland) [reply indykite/resource_external_data_resolver.go:82]: the design doc doesn't address it specifically, but I thought the plan for phase 1 was to just support `GET`. @arxeiss @jakubtomany ?

Comment (Septimus4) [reply indykite/resource_external_data_resolver.go]: I am allowing GET only and if we need to add more we will make terraform PR's

Comment (arxeiss) [reply indykite/resource_external_data_resolver.go]: We should update design doc then. Supporting only GET is not enough as you cannot send payload with it.

So yes, we support all those mentioned.

File: indykite/resource_consent.go (hunk 1 above)
Comment (arxeiss) [reply indykite/resource_consent.go:157]: This is wrong code and should be removed. Terraform always returns numbers in `int` so other cases would never happen.

File: indykite/resource_external_data_resolver.go (hunk 3 above)
Comment (arxeiss) [reply indykite/resource_external_data_resolver.go]: OK, so I checked https://www.notion.so/indykite/Data-Reference-external-data-Technical-Design-Phase-1-d81af130e76d4b85a251e36b8cf37ad9 and there is no mention about supported/unsupported HTTP Methods. cc @maaland. So I would expect a fix by @Septimus4

Comment (maaland) [reply indykite/resource_external_data_resolver.go]: thanks for clarifying
